src/key_packages/mod.rs

```diff
@@ -72,11 +71,11 @@ impl KeyPackage {
             protocol_version: ProtocolVersion::default(),
             cipher_suite: ciphersuite_name,
             hpke_init_key,
-            credential,
+            credential: credential_bundle.credential().clone(),
```

Comment:
Let's not pass in a reference. Then we don't have to clone. The caller is then responsible to clone if they need the credential again.

Reply:
This is tricky, because the `credential_bundle` should not be part of the `key_package`, but the `credential` should be. In `key_packages` owned by other parties, we need to be able to update these `key_package`s, as they might change over time and also might not be the same in every group for that party. In `key_package`s belonging to ourselves, I'm not sure if we want to attach the `credential_bundle`, because we might have one for multiple groups, but then again, we might not.

I don't know what's the best way to do ownership here, but I think we should clear this up in a separate PR.

Reply:
Filed issue #161.